### PR TITLE
#0: added normalization details in the tech report

### PR DESCRIPTION
**Pull request by kpaigwar**
### Problem description
Added Distributed and Non-Distributed normalization details in LLM tech report.

### CODE REVIEW

File: tech_reports/LLMs/llms.md [hunk 1]
```diff
@@ -43,8 +43,154 @@ Authors:
...
+
+TT-NN includes two primary implementations of normalization operations to handle diverse activation layouts efficiently:
+
+1. **Non-Distributed Norm**
+2. **Distributed Norm**
+
+
+#### 1. Non-Distributed Norm
+
+This implementation supports both sharded and interleaved inputs. It is employed in the following scenarios:
```
Comment (uaydonat) [inline tech_reports/LLMs/llms.md:59]: What does it mean to be non-distributed? Needs to be mentioned

File: tech_reports/LLMs/llms.md [hunk 2]
```diff
@@ -43,8 +43,154 @@ Authors:
...
+    memory_config=ttnn.DRAM_MEMORY_CONFIG
+)
+
+# Perform RMSNorm
+ttnn_output = ttnn.rms_norm(ttnn_input, epsilon=1e-5, weight=ttnn_gamma)
+```
+
+**Optimization for Efficient Weight Reads from DRAM**
+
+In above example, weights were traditionally pushed to device in **TILE layout**. But in this case, padding is required to match the TILE_HEIGHT. This padding increased memory footprint and reduced DRAM access efficiency. To address this, weights are now wrapped into **TILE_WIDTH** sticks and converted to **ROW_MAJOR_LAYOUT** without requiring any padding.
```
Comment (uaydonat) [inline tech_reports/LLMs/llms.md:105]: Is there a perf cost for this transformation? Maybe mention that.

File: tech_reports/LLMs/llms.md [hunk 3]
```diff
@@ -43,8 +43,154 @@ Authors:
...
+- **Interleaved Inputs**:
+  For interleaved inputs, the kernel parallelizes work across the sequence length (`seq_len`).
+  This makes it highly **optimal for prefill cases**, where the sequence length is large.
+
+- **Width-Sharded Inputs**:
+  For width-sharded inputs, the kernel splits the work across the embedding dimension.
+  This design is more **optimal for decode cases**, where the sequence length is typically `seq_len=1`.
+
+
+#### References
```
Comment (uaydonat) [inline tech_reports/LLMs/llms.md:187]: Cool. I liked the references to the code.

**Review by uaydonat — CHANGES_REQUESTED** [review]

File: tech_reports/LLMs/llms.md (hunk 1 above)
Comment (kpaigwar) [reply tech_reports/LLMs/llms.md]: Thanks, addressed it!

File: tech_reports/LLMs/llms.md (hunk 2 above)
Comment (kpaigwar) [reply tech_reports/LLMs/llms.md]: There is no perf cost at runtime as its done once during model initialization. Added that

File: tech_reports/LLMs/llms.md [hunk 4]
```diff
@@ -56,8 +57,154 @@ Other useful resources:
...
+
+TT-NN includes two primary implementations of normalization operations to handle diverse activation layouts efficiently:
+
+1. **Non-Distributed Norm**
+2. **Distributed Norm**
+
+
+#### 1. Non-Distributed Norm
+
+**Non-Distributed Norm** refers to the standard implementation of normalization operations applied to activations that are not distributed across multiple devices. This implementation supports both sharded and interleaved inputs. It is employed in the following scenarios:
```
Comment (uaydonat) [inline tech_reports/LLMs/llms.md:73]: you mean activations are stored in a single device? Maybe this is clearer.

Comment (kpaigwar) [reply tech_reports/LLMs/llms.md]: Yeah but I wanted to stress that they arenot distributed but can also be replicated across multiple devices.
I modified to below, i hope its more clearer.
_Non-Distributed Norm refers to the standard implementation of normalization operations applied to activations that are not distributed across multiple devices. This type of normalization is suitable for setups where the entire activation or embedding is available locally on a single device or is replicated identically across multiple devices in a data-parallel setup._

**Review by uaydonat — APPROVED** [review]

**Pull request merged**